### PR TITLE
Test Engine Upgrade

### PR DESCRIPTION
**Pull request by evanjtravis**
With the unit-tests passing, we need to review the sanity of the feature files before merging into master.

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -0,0 +1,15 @@
+--extra-index-url https://pip-test.techservices.illinois.edu/index/test
```
Comment (ddriddle) [inline requirements.txt]: What is this file for? Why is there not a requirements.txt file in https://github.com/techservicesillinois/terraform-aws-ecr/pull/7?

File: test/features/ecs-service/ecs-services.feature [hunk 2]
```diff
@@ -95,7 +95,7 @@ Feature: We are able to instantiate all aws_ecs_service resources
             #--------|-------------------------------|---------------------|-------|
             | create | aws_alb_listener_rule         | set_priority        |       |
             |        | aws_ecs_service               | awsvpc_all          |       |
-            |        | aws_ecs_task_definition       | default             |       |
+            |        | aws_ecs_task_definition       | fargate             |       |
```
Comment (ddriddle) [inline test/features/ecs-service/ecs-services.feature:98]: Why was this change necessary?

**Review by ddriddle — CHANGES_REQUESTED** [review]
This branch is out of date. Please:

- update your fork and rebase your branch so we have  a clean diff.
- remove the test/README.md. We agreed not to have this copied around into each repo.

File: requirements.txt (hunk 1 above)
Comment (evanjtravis) [reply requirements.txt]: Probably just an oversight when I was trying to get the test environment working. It's not used in the workflow or in the installation process. I'll get rid of it.

File: test/features/ecs-service/ecs-services.feature (hunk 2 above)
Comment (evanjtravis) [reply test/features/ecs-service/ecs-services.feature:98]: It was necessary only because I couldn't figure out why the test was failing with the "default" name. I figured the change occurred with the 0.12 upgrade of the terraform code, but this may warrant further investigation.

**Comment by ddriddle** [comment]
@JonRoma I was able to run the `ecs-service` tests on my laptop successfully:

```
$ git clone -b feature/unit-tests git@github.com:evanjtravis/terraform-aws-ecs-service.git
$ cd terraform-aws-ecs-service/test/
$ make
...
**************************************************
* EXPECTED BEHAVIOR OBSERVED
**************************************************

2 features passed, 0 failed, 0 skipped
6 scenarios passed, 0 failed, 0 skipped
68 steps passed, 0 failed, 0 skipped, 0 undefined
Took 15m30.973s
```

@evanjtravis again excellent work! Thank you very much. @JonRoma please verify that you able to run the tests.

File: README.md [hunk 3]
```diff
@@ -515,6 +517,16 @@ A `service_discovery` block configures [Amazon Service Discovery](https://docs.a
 
 * `namespace_id` – (Required) The ID of the namespace to use for DNS configuration.
 
+* `container_name` - (Required) The name of the container to use
```
Comment (ddriddle) [inline README.md]: The service discovery map does not support container_name as input.

File: README.md [hunk 4]
```diff
@@ -515,6 +517,16 @@ A `service_discovery` block configures [Amazon Service Discovery](https://docs.a
...
+with the service discovery service as it appears in the container
+definition (by default, `containers.json`). Defaults to the value
+specified in the [Load Balancer block](#load_balancer).
+
+* `container_port` - (Required) The port on the container to use
```
Comment (ddriddle) [inline README.md]: The service discovery map does not support container_port as input.

File: README.md [hunk 5]
```diff
@@ -352,12 +352,14 @@ A `load_balancer` block supports the following:
 * `port` - (Optional) The port of the listener. Defaults to 443.
 
 * `container_name` - (Required) The name of the container to associate
-with the load balancer as it appears in the container definition (by
-default, `containers.json`).
+with the load balancer as it appears in the container definition
+(by default, `containers.json`). Defaults to the value specified
```
Comment (ddriddle) [inline README.md]: This is false as the code is written.

File: README.md [hunk 6]
```diff
@@ -352,12 +352,14 @@ A `load_balancer` block supports the following:
...
+in the [Service Discovery](#service_discovery) block.
 
 * `container_port` - (Required) The port on the container to associate
-with the load balancer as it appears in the container definition (by
-default, `containers.json`).
+with the load balancer as it appears in the container definition
+(by default, `containers.json`). Defaults to the value specified
```
Comment (ddriddle) [inline README.md]: This is false as the code is written.

File: test/features/ecs-service/ecs-services.feature (hunk 2 above)
Comment (ddriddle) [reply test/features/ecs-service/ecs-services.feature:98]: This is correct after reviewing the Terraform code.

**Review by ddriddle — CHANGES_REQUESTED** [review]
@evanjtravis please revert all changes to the README.md. Please update the Makefile & Dockerfile per changes recommended for as-aws-lb.

File: README.md (hunk 3 above)
Comment (evanjtravis) [reply README.md]: I will remove this line from the file.

File: README.md (hunk 5 above)
Comment (evanjtravis) [reply README.md]: Directions unclear. What should the change be to reflect reality? Remove `container_name` variable from the `load_balancer` section?

File: README.md (hunk 6 above)
Comment (evanjtravis) [reply README.md]: Directions unclear. What should the change be to reflect reality? Remove `container_port` variable from the `load_balancer` section?

File: README.md (hunk 5 above)
Comment (ddriddle) [reply README.md]: Please revert all changes to the README.md. I don't know how or why any of these changes are present. All the changes are incorrect. Revert the README.md to what is found in `master`.

File: README.md (hunk 4 above)
Comment (evanjtravis) [reply README.md]: Removed.

**Review by ddriddle — APPROVED** [review]
Looks good!

**Comment by ddriddle** [comment]
@evanjtravis @JonRoma the `ecs-service` test as written has a dependency on the `examples` CloudWatch log group. This dependency will be removed by the next card so I'm going to ignore this issue.

**Review by ddriddle — APPROVED** [review]
The tests work.
